graylog2-server/src/main/java/org/graylog2/security/ldap/LdapSettingsImpl.java

```diff
@@ -238,4 +279,116 @@ public void setTrustAllCertificates(boolean trustAllCertificates) {
         fields.put(TRUST_ALL_CERTS, trustAllCertificates);
     }
 
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    @Override
+    public Map<String, String> getGroupMapping() {
+        final Map<String, String> groupMapping = (Map<String, String>) fields.get(GROUP_MAPPING);
+
+        if (groupMapping == null || groupMapping.isEmpty()) {
+            return Collections.emptyMap();
+        }
+        else {
+            // we store role ids, but the outside world uses role names to identify them
+            try {
+                final Map<String, Role> idToRole = roleService.loadAllIdMap();
+                return Maps.newHashMap(Maps.transformValues(groupMapping, Roles.roleIdToNameFunction(idToRole)));
+            } catch (NotFoundException e) {
+                LOG.error("Unable to load role mapping");
+                return Collections.emptyMap();
+            }
+        }
+    }
+
+    @Override
+    public void setGroupMapping(Map<String, String> mapping) {
+        Map<String, String> internal;
+        if (mapping == null) {
+            internal = Collections.emptyMap();
+        } else {
+            // we store ids internally but external users use the group names
+            try {
+                final Map<String, Role> nameToRole = Maps.uniqueIndex(roleService.loadAll(), Roles.roleToNameFunction());
+
+                internal = Maps.newHashMap(Maps.transformValues(mapping, new Function<String, String>() {
+                    @Nullable
+                    @Override
+                    public String apply(@Nullable String groupName) {
+                        if (groupName == null || !nameToRole.containsKey(groupName)) {
+                            return null;
+                        }
+                        return nameToRole.get(groupName).getId();
+                    }
+                }));
+            } catch (NotFoundException e) {
+                LOG.error("Unable to convert group names to ids", e);
+                throw new IllegalStateException("Unable to convert group names to ids", e);
+            }
+        }
+
+        fields.put(GROUP_MAPPING, internal);
+    }
+
+    @Override
+    public String getGroupSearchBase() {
+        final Object o = fields.get(GROUP_SEARCH_BASE);
+        return o != null ? o.toString() : "";
+    }
+
+    @Override
+    public void setGroupSearchBase(String groupSearchBase) {
+        fields.put(GROUP_SEARCH_BASE, groupSearchBase);
+    }
+
+    @Override
+    public String getGroupIdAttribute() {
+        final Object o = fields.get(GROUP_ID_ATTRIBUTE);
+        return o != null ? o.toString() : "";
```

Comment:
Minor nitpick: Use [`MoreObjects#firstNonNull(T, T)`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/MoreObjects.html#firstNonNull(T,%20T))

Alternatively `return nullToEmpty((String) o);` should work.